### DISCUSSION TITLE
hoe tsp works

### DISCUSSION THREAD

**Discussion by manwithmostash** [vedanthvnayak/tsp_visual_demo#6] [Q&A]
how actually tsp works

**Comment by vedanthvnayak** [comment]
Given a set of cities and the distance between every pair of cities, the problem is to find the shortest possible route that visits every city exactly once and returns to the starting point. Note the difference between [Hamiltonian Cycle](https://www.geeksforgeeks.org/backtracking-set-7-hamiltonian-cycle/) and TSP. The Hamiltonian cycle problem is to find if there exists a tour that visits every city exactly once. Here we know that Hamiltonian Tour exists (because the graph is complete) and in fact, many such tours exist, the problem is to find a minimum weight Hamiltonian Cycle.
